### ISSUE TITLE
DRCSim Compile Instructions Need to be updated for DRCSim 2.0

### ISSUE THREAD

**Issue by osrf-migration** [osrf/drcsim#120]
**Original report ([archived issue](https://osrf-migration.github.io/drcsim-gh-pages/#!/osrf/drcsim/issues/120)) by John Hoare (Bitbucket: [jhoare](https://bitbucket.org/%7Bf2fb0aad-9933-4b16-a710-d9fe1486032f%7D/)).**

----------------------------------------
The instructions for compiling DRCSim from source are not up to date for DRCSim 2.0. 

http://gazebosim.org/wiki/DRC/Install

**Comment by osrf-migration** [comment]
**Original comment by Jesper Smith (Bitbucket: [jespersmith](https://bitbucket.org/%7B61d9fe7d-e554-408c-99fb-faefa9c96114%7D/)).**

----------------------------------------
Just got it to work on Ubuntu 12.10. Here are my notes

Installing ROS Groovy

Follow the instructions on http://www.ros.org/wiki/groovy/Installation/Ubuntu and do the Desktop-Full Install

The short version is

     sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu quantal main" > /etc/apt/sources.list.d/ros-latest.list'

    wget http://packages.ros.org/ros.key -O - | sudo apt-key add -

    sudo apt-get update

    sudo apt-get install ros-groovy-desktop-full

Installing build dependencies

    sudo apt-get install build-essential cmake libfreeimage-dev libprotoc-dev\
                         libprotobuf-dev protobuf-compiler freeglut3-dev libcurl4-openssl-dev \
                         libtinyxml-dev libtar-dev libtbb-dev libogre-dev \
                         libxml2-dev pkg-config libqt4-dev ros-fuerte-urdfdom libltdl-dev libboost-thread-dev \
                         libboost-signals-dev libboost-system-dev libboost-filesystem-dev libboost-program-options-dev \
                         libboost-regex-dev libboost-iostreams-dev debhelper mercurial

Installing orsf-common

This requires some magic.

Check out osrf-common

    hg clone #!/osrf/osrf-common osrf-common

Setup build envirionment

    source /opt/ros/groovy/setup.bash

    export ROS_PACKAGE_PATH=$ROS_PACKAGE_PATH:`pwd`/osrf-common/ros

    cd osrf-common

    mkdir build

Compile osrf-common

    cd build

    cmake -DCMAKE_INSTALL_PREFIX=/opt/ros/groovy ../

    make

    sudo make install

Installing sandia-hand

This also requires some magic

Check out sandia-hand

    hg clone #!/osrf/sandia-hand sandia-hand

Setup build envirionment

    source /opt/ros/groovy/setup.bash

    export ROS_PACKAGE_PATH=$ROS_PACKAGE_PATH:`pwd`/sandia-hand/ros

    cd sandia-hand

    mkdir build

Compile sandia-hand

    cd build

    cmake -DCMAKE_INSTALL_PREFIX=/opt/ros/groovy ../

    make

    sudo make install

Installing Gazebo

Short version

    hg clone #!/osrf/drcsim

    cd drcsim

    mkdir build

    cd build

    source /opt/ros/groovy/setup.bash

    cmake -DCMAKE_INSTALL_PREFIX=/usr ..

    make

    make install

Installing DRCSim

Short version

    hg clone #!/osrf/drcsim

    cd drcsim

    mkdir build

    cd build

    source /opt/ros/groovy/setup.bash

    source /usr/share/gazebo/setup.sh

    cmake -DCMAKE_INSTALL_PREFIX=/usr ..

    make

    make install

Running DRCSim

Add "source /usr/share/drcsim/setup.sh" to ~/.bashrc and start a new terminal

No hands

roslaunch atlas_utils atlas.launch

Sandia hands

roslaunch atlas_utils atlas_sandia_hands.launch

iRobot hands

roslaunch atlas_utils atlas_irobot_hands.launch

**Comment by osrf-migration** [comment]
**Original comment by Brian Gerkey (Bitbucket: [Brian Gerkey](https://bitbucket.org/%7B3d8b5763-9dad-4be7-85c2-a7ab6cd49173%7D/), GitHub: [gerkey](https://github.com/gerkey)).**

----------------------------------------

* changed state from "new" to "resolved"

I believe that the current instructions are correct.  Please re-open if not.

**Comment by osrf-migration** [comment]
**Original comment by John Hoare (Bitbucket: [jhoare](https://bitbucket.org/%7Bf2fb0aad-9933-4b16-a710-d9fe1486032f%7D/)).**

----------------------------------------

* changed state from "resolved" to "open"

They are not up to date. What Jesper has is pretty accurate, but could use some cleaning up. The guide makes no mention of compiling osrf-common or sandia-hand

**Comment by osrf-migration** [comment]
**Original comment by Brian Gerkey (Bitbucket: [Brian Gerkey](https://bitbucket.org/%7B3d8b5763-9dad-4be7-85c2-a7ab6cd49173%7D/), GitHub: [gerkey](https://github.com/gerkey)).**

----------------------------------------

* changed state from "open" to "resolved"

I updated the install instructions to also describe building osrf-common and sandia-hand.  I've only tested them on Precise/Fuerte, but I would expect them to work (with some substitutions) on Quantal/Groovy.
